README.md

```diff
@@ -6,9 +6,37 @@ Another Validator of OpenAPI spec repository Configuration And Directories.
 
 NPM: https://www.npmjs.com/package/@azure/avocado
 
+## Overview
+
+Avocado validate folder structure and configuration for repository "Azure/azure-rest-api-specs". Avocado report new errors involved in PR. Avocado is a CI tool that will trigger automatically by azure DevOps pipeline when a new pull request is created.
+
+NOTE: When running in azure devops Avocado only report new errors involved in PR, but ignore the previous existing errors. When running in local machine, Avocado report all errors.
+
+Avocado major functions are listed below:
+
+- For a given directory validate whether exists `specification` and filter `readme.md` under the `specification` folder.
+- Validate whether `readme.md` is autorest specific file which must contain `see https://aka.ms/autorest`
+- Validate all `swagger file` whether is correct parsed by `json`, and check all referenced `json` file (`referenced json` file marked in json object has the key name `"$ref"`)
+- Validate whether the folder has no referenced file. `swagger file` must be referenced by `readme.md` or other `swagger file`.
+- Validate whether `swagger file` has a circular reference and report a warning.

```

Comment:
add an example of circular reference.

Reply:
Add section `How to solve errors`. This section now only contains CIRCULAR_REFERENCE. I will add other errors here later.